### DISCUSSION TITLE
Is there a way to get all keys and values for a limiter?

### DISCUSSION THREAD

**Discussion by mauriciocirelli** [animir/node-rate-limiter-flexible#171] [Ideas]
Dear,

I would like to create a view in my admin website showing all keys and their values.
Currently, using redis, I have an option with redisScan to do that..
But it would be great if we could do that directly on the rateLimiter object.

Is there any way to do it?

Thank you!

**Comment by animir** [comment]
@mauriciocirelli Hey, there is no such functionality now. Could you share your code with `redisScan`, when you implement it?

**Reply by mauriciocirelli** [reply]
Sure, it is pretty simple. It uses a callback function when completed. Could easily use promises also.

```
function scan(prefix, cb)
{
    var objs = [];
    redisScan({
        redis: redisClient,
        pattern: prefix + "*",
        each_callback: function (type, key, subkey, dummy, value, next) {
            objs.push({ key: key, value: JSON.parse(value) });
            next();
        },
        done_callback: function (err) {
            if(cb)
                cb(err, objs);
        }
    });
}
```
It would be great if we had something like this for any kind of limiter.

**Reply by animir** [reply]
I put it to Wiki, thanks.
